jest.config.js

```diff
@@ -1,5 +1,27 @@
-const { compilerOptions } = require('./tsconfig.json');
+const tsConfig = require('get-tsconfig').getTsconfig();
+const compilerOptions = tsConfig.config.compilerOptions;
+const { baseUrl, paths } = compilerOptions;
+const useSwc = !tsConfig.path.includes('packages/cli');
 
+/** @type {import('@swc/types').Config} */
+const swcJestOptions = {
+	jsc: {
+		parser: {
+			syntax: 'typescript',
+			tsx: false,
+			decorators: true,
+		},
+		transform: {
+			legacyDecorator: true,
+			decoratorMetadata: true,
+		},
+		target: 'es2020',
```

Comment:
Should we take this value from the tsconfig instead of hardcoding it?

Reply:
we could, but I was assuming that newer targets would mean faster transpilation in tests. but, I actually haven't benchmarked that.